### PR TITLE
Limiting team/repository access for users inactive for 60 months

### PR DESCRIPTION
**Pull request by galargh**
### Summary

This PR cleans up user access by removing users who have been inactive for over 60 months (5 years) from teams and repositories.

A user is deemed inactive if they haven't performed any of the following actions in the past 60 months in a repository in question or in any of the repositories the team in question grants access to:
- created an issue or a pull request
- commented on an issue or a pull request
- reviewed a pull request
- commented on a commit
- pushed (in particular, force pushed) to a branch
- created a branch
- deleted a branch
- merged a pull request
- added a pull request to a merge queue

Any user who, after the introduction of the above changes, isn't a direct collaborator in any of the repositories and isn't a member of any teams is assigned to the `Alumni` team.

If a user's access to a repository or team should be restored, the appropriate line change should be reverted, and a comment starting with `KEEP: ` (followed by a reason) should be added directly above that line.

This pertains to the "'archive' inactive users and teams" in https://github.com/ipfs/ipfs/issues/511.

### Who is this targeting?

The current PR is what results from a [script to identify inactive users in an org](https://github.com/ipfs-shipyard/github-mgmt/blob/master/scripts/src/actions/remove-inactive-members.ts).

### Why is this being done?
See "Why do we care about periodically cleaning up permissions across the orgs?" in https://github.com/ipfs/ipfs/issues/511

### Is this set in stone?
No. This PR was created and being left open for some days to give awareness and incorporate feedback. We're not taking a "ask for permission" approach, as that would require way too much wrangling. Instead, we're giving visibility to what's proposed and inviting folks to comment and influence. A saving grace here is that none of this is a "one-way door". If something got messed up or missed, a follow-up PR can be done to correct it.

### Is anyone being removed from the organization?
No. All existing members of the org are staying members. In the most reduced/scoped-down case, someone will still be part of an "Alumni" team in the org to signal their past involvement. Thank you for your past contributions, and we certainly welcome you to play a more active role in the future.

### Timeline
2024-02-26: public PR
2024-03-04: notify affected parties with @mention: <insert LINK_TO_COMMENT_IN_THIS_PR>
2024-03-08: merge this change after incorporating feedback
2024-03-08: remove empty teams

### CODE REVIEW

File: github/ipfs-shipyard.yml [hunk 1]
```diff
@@ -695,9 +679,7 @@ repositories:
     archived: false
     collaborators:
       admin:
-        - mishmosh
```
Comment (lidel) [inline github/ipfs-shipyard.yml:698]: `# KEEP:`

File: github/ipfs-shipyard.yml [hunk 2]
```diff
@@ -569,7 +554,6 @@ repositories:
     archived: false
     collaborators:
       admin:
-        - autonome
```
Comment (lidel) [inline github/ipfs-shipyard.yml:572]: `# KEEP`

File: github/ipfs-shipyard.yml [hunk 3]
```diff
@@ -4197,7 +4085,6 @@ repositories:
...
```
Comment (lidel) [inline github/ipfs-shipyard.yml:4200]: `# KEEP: leading https://areweinterplanetaryyet.org`

File: github/ipfs-shipyard.yml [hunk 4]
```diff
@@ -4666,29 +4550,16 @@ teams:
     description: People with write access to Javascript repositories
```
Comment (lidel) [inline github/ipfs-shipyard.yml:4550]: :thought_balloon:  I don't think we've been using this group for day-to-day, superseded by [helia-dev](https://github.com/orgs/ipfs-shipyard/teams/helia-dev)

**Review by lidel — COMMENTED** [review]
Similar to https://github.com/ipfs/github-mgmt/pull/193#pullrequestreview-1911844421, nothing controversial stands out, removing permissions from inactive members makese sense.

Realization: we have a lot of dead teams.

File: github/ipfs-shipyard.yml (hunk 4 above)
Comment (galargh) [reply github/ipfs-shipyard.yml:4550]: The current version of a script assumes that a team is active as long as any user in that team remains active in at least one repository that team grants access to. So if we know of some teams that should not exist anymore, I think it'll be more efficient to flag and remove them proactively.

**Comment by galargh** [comment]
Hi! We're ready to open up the PR for general review 🥳

I'd like to ask you to review the changes affecting you and flag any that should be reverted, are wrong, or need more explanation. You can find the detailed explanation of this PR, the reasoning for introducing the changes and the process itself in the description - https://github.com/ipfs-shipyard/github-mgmt/pull/77#issue-2153562115

Thank you, and let me know if you have any questions 💁

Tagging all the people whose access changes (https://github.com/ipfs-shipyard/github-mgmt/pull/77#issuecomment-1963562925) as a result of this PR (no one is being removed from the org):
- [ ] @2color
- [ ] @Areadrill
- [ ] @AuHau
- [ ] @BigLep
- [ ] @Gozala
- [ ] @Kubuxu
- [ ] @Ricardo-Silva91
- [ ] @RichardLitt
- [ ] @SgtPooki
- [ ] @achingbrain
- [ ] @akrych
- [ ] @alanshaw
- [ ] @andyschwab
- [ ] @b5
- [ ] @berty-assistant
- [ ] @bmann
- [ ] @cewood
- [ ] @cwaring
- [ ] @davidd8
- [ ] @daviddias
- [ ] @dignifiedquire
- [ ] @dirkmc
- [ ] @gammazero
- [ ] @glouvigny
- [ ] @guseggert
- [ ] @haadcode
- [ ] @hsanjuan
- [ ] @hugomrdias
- [ ] @ipfs-gui-bot
- [ ] @ipfsbot
- [ ] @jacobheun
- [ ] @jdiogopeixoto
- [ ] @jimpick
- [ ] @jnthnvctr
- [ ] @juliaxbow
- [ ] @krl
- [ ] @lidel
- [ ] @locotorp
- [ ] @magik6k
- [ ] @marcooliveira
- [ ] @mishmosh
- [ ] @momack2
- [ ] @moul-sudo
- [ ] @nlfonseca
- [ ] @ntninja
- [ ] @olizilla
- [ ] @parkan
- [ ] @petar
- [ ] @ruimonteiro93
- [ ] @rvagg

1/2

**Comment by galargh** [comment]
- [ ] @stongo
- [ ] @timelytree
- [ ] @vasco-santos
- [ ] @vesahc
- [ ] @victorb
- [ ] @vmx
- [ ] @willscott
- [ ] @yusefnapora

2/2

File: github/ipfs-shipyard.yml [hunk 5]
```diff
@@ -2233,12 +2175,8 @@ repositories:
       admin:
         - autonome
         - cwaring
-        - vesahc
       push:
-        - b5
-        - bmann
         - mishmosh
-        - SgtPooki
```
Comment (SgtPooki) [inline github/ipfs-shipyard.yml:2241]: Good with this

File: github/ipfs-shipyard.yml [hunk 6]
```diff
@@ -2805,8 +2740,6 @@ repositories:
         - Gozala
         - SgtPooki
         - whizzzkid
-      push:
-        - ipfs-gui-bot
```
Comment (SgtPooki) [inline github/ipfs-shipyard.yml:2809]: I dont think we need this bot anymore, i can make sure this repo is configured with unifiedCI properly if i come across it again

File: github/ipfs-shipyard.yml [hunk 7]
```diff
@@ -3129,9 +3060,6 @@ repositories:
     advanced_security: false
     allow_update_branch: false
     archived: false
-    collaborators:
-      admin:
-        - achingbrain
```
Comment (SgtPooki) [inline github/ipfs-shipyard.yml:3134]: I think Alex was the primary dev here but im sure he'd like less on his plate

File: github/ipfs-shipyard.yml [hunk 8]
```diff
@@ -4599,29 +4499,16 @@ teams:
     description: People with write access to Javascript repositories
     members:
       maintainer:
-        - 2color
         - lidel
-        - rvagg
-        - sgtpooki
```
Comment (SgtPooki) [inline github/ipfs-shipyard.yml:4605]: Im not a fan of being removed here. Will I still have write permissions?

File: github/ipfs-shipyard.yml [hunk 9]
```diff
@@ -4599,29 +4499,16 @@ teams:
...
         - hacdias
         - hsanjuan
         - hugomrdias
-        - ipfsbot
         - jacobheun
         - jbenet
-        - magik6k
-        - momack2
-        - olizilla
-        - vasco-santos
```
Comment (SgtPooki) [inline github/ipfs-shipyard.yml:4622]: Add me here maybe?

**Review by SgtPooki — CHANGES_REQUESTED** [review]
Only a single flag for me: javascript group write (probably for `achingbrain` too)

File: github/ipfs-shipyard.yml (hunk 8 above)
Comment (galargh) [reply github/ipfs-shipyard.yml:4605]: As per https://github.com/ipfs-shipyard/github-mgmt/pull/77#issuecomment-1963562925, here's your entire access changeset:

```
User SgtPooki:
  - will lose push permission to ipfs-thing-2023
```

This suggests to me `Javascript` team is not used much after all.

Comment (SgtPooki) [reply github/ipfs-shipyard.yml:4605]: Nah its probably fine

**Pull request merged**